src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -195,17 +213,44 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
                          location, onboarding_tenant_id, http_proxy, https_proxy, no_proxy, private_key_pem, kube_config,
-                         kube_context, no_wait)
+                         kube_context, no_wait, values_file_provided, values_file)
 
     return put_cc_response
 
 
-def validate_cloud(cmd):
-    if cmd.cli_ctx.cloud.endpoints.resource_manager == consts.Dogfood_RMEndpoint:
+def send_cloud_telemetry(cmd):
+    cloud_name = cmd.cli_ctx.cloud.name
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.AzureCloud': cloud_name})
+
+
+def validate_env_file_dogfood(values_file, values_file_provided):
+    if not values_file_provided:
         telemetry.set_user_fault()
-        telemetry.set_exception(exception='Dogfood cloud not supported.', fault_type=consts.Load_Kubeconfig_Fault_Type,
-                                summary='Dogfood cloud not supported.')
-        raise CLIError("Connectedk8s CLI is not supported for Dogfood environment. Please switch the cloud using 'az cloud set --name {cloudName}' and try again. For Dogfood cloud, use helm directly for onboarding.")
+        telemetry.set_exception(exception='Helm environment file not provided', fault_type=consts.Helm_Environment_File_Fault_Type,
+                                summary='Helm environment file missing')
+        raise CLIError("Helm environment file is required when using Dogfood environment for onboarding the cluster. Please set the environment variable 'HELMVALUESPATH' to point to the file.")
+
+    with open(values_file, 'r') as f:
+        try:
+            env_dict = yaml.safe_load(f)
+        except Exception as e:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception=e, fault_type=consts.Helm_Environment_File_Fault_Type,
+                                    summary='Problem loading the helm environment file')
+            raise CLIError("Problem loading the helm environment file: " + str(e))
+        try:
+            assert env_dict.get('global').get('azureEnvironment') == 'AZUREDOGFOOD'
+            assert env_dict.get('systemDefaultValues').get('azureArcAgents').get('config_dp_endpoint_override')
+        except Exception as e:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception=e, fault_type=consts.Helm_Environment_File_Fault_Type,
+                                    summary='Problem loading the helm environment variables')
+            raise CLIError("The required helm environment variables for dogfood onboarding are either not present in the file or incorrectly set. Please check the values 'global.azureEnvironment' and 'systemDefaultValues.azureArcAgents.config_dp_endpoint_override' in the file.")
+
+    # Return the dp endpoint and release train
+    dp_endpoint = env_dict.get('systemDefaultValues').get('azureArcAgents').get('config_dp_endpoint_override')
+    release_train = env_dict.get('systemDefaultValues').get('azureArcAgents').get('releaseTrain')
```

Comment:
>releaseTrain [](start = 83, length = 12)

what if this is nont set?

Reply:
Then default will be stable. That is because for config dp endpoint (for dogfood), setting release train to stable fetches the dev chart and setting release train to dev causes error

Reply:
What would this command return if i dont have field releasetrain in yaml
env_dict.get('systemDefaultValues').get('azureArcAgents').get('releaseTrain')


---
In reply to: [478991795](https://github.com/Azure/azure-cli-extensions/pull/2238#discussion_r478991795) [](ancestors = 478991795)

Reply:
It will be 'None'